pkg/nfd-worker/nfd-worker.go

```diff
@@ -713,25 +729,6 @@ func (m *nfdWorker) updateNodeFeatureObject(labels Labels) error {
 
 	features := source.GetAllFeatures()
 
-	// Create owner ref
-	ownerRefs := []metav1.OwnerReference{}
-	podName := os.Getenv("POD_NAME")
```

Comment:
I think we want to keep this owner ref as well. I.e. have both the DS (or whatever there are) and the Pod. NF will be deleted only when all the owners are gone.

Reply:
done